recipes/xtr/all/conanfile.py

```diff
@@ -1,125 +1,159 @@
+from conan.tools.layout import basic_layout
+from conan.tools.scm import Version
+
+required_conan_version = ">=1.53.0"
+
 
 class XtrConan(ConanFile):
     name = "xtr"
-    description = \
-        "C++ Logging Library for Low-latency or Real-time Environments"
-    topics = ("xtr", "logging", "logger")
+    description = "C++ Logging Library for Low-latency or Real-time Environments"
+    license = "MIT"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/choll/xtr"
-    license = "MIT"
+    topics = ("logging", "logger")
+
+    package_type = "library"
     settings = "os", "arch", "compiler", "build_type"
     options = {
+        "shared": [True, False],
         "fPIC": [True, False],
         "enable_exceptions": [True, False],
         "enable_lto": [True, False],
         "enable_io_uring": ["auto", True, False],
         "enable_io_uring_sqpoll": [True, False],
-        "sink_capacity_kb": "ANY"
+        "sink_capacity_kb": [None, "ANY"],
     }
     default_options = {
+        "shared": False,
         "fPIC": True,
         "enable_exceptions": True,
         "enable_lto": False,
         "enable_io_uring": "auto",
         "enable_io_uring_sqpoll": False,
-        "sink_capacity_kb": None
+        "sink_capacity_kb": None,
     }
 
-    generators = "make"
-
     def config_options(self):
-        if tools.Version(self.version) < "1.0.1":
-            del self.options.sink_capacity_kb
-        if tools.Version(self.version) < "2.0.0":
+        if Version(self.version) < "2.0.0":
             del self.options.enable_io_uring
             del self.options.enable_io_uring_sqpoll
 
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
     def requirements(self):
-        self.requires("fmt/7.1.3")
+        # INFO: https://github.com/choll/xtr/blob/2.1.0/include/xtr/detail/buffer.hpp#L27
+        self.requires("fmt/10.1.1", transitive_headers=True, transitive_libs=True)
         # Require liburing on any Linux system as a run-time check will be
         # done to detect if the host kernel supports io_uring.
-        if tools.Version(self.version) >= "2.0.0" and self.settings.os == "Linux" and self.options.get_safe("enable_io_uring"):
-            self.requires("liburing/2.1")
+        if (
+            Version(self.version) >= "2.0.0"
+            and self.settings.os in ["Linux", "FreeBSD"]
+            and self.options.get_safe("enable_io_uring")
+        ):
+            self.requires("liburing/2.4")
 
     def validate(self):
-        if self.settings.os not in ("FreeBSD", "Linux"):
-            raise ConanInvalidConfiguration(f"Unsupported os={self.settings.os}")
-        if self.settings.compiler not in ("gcc", "clang"):
-            raise ConanInvalidConfiguration(f"Unsupported compiler={self.settings.compiler}")
-        if self.settings.arch not in ("x86_64", ):
```

Comment:
xtr contains some x86 inline assembly so I think it would be better to keep the arch check

Reply:
Yeah. It won't work on macOS or Windows either, FYI. It might work on other varieties of Unix but is only supported/tested on Linux and FreeBSD at the moment.

Reply:
It's just x86_64 at the moment

Reply:
Ok, thanks! And thanks for taking the time to review the recipe as well!

Reply:
np, thanks for updating the recipe. Are you using xtr?